icefoganalytics/travel-authorization
# Standardize `user` Table

Fixes https://github.com/icefoganalytics/travel-authorization/issues/66

# Context

See https://github.com/icefoganalytics/travel-authorization/issues/30
See https://github.com/icefoganalytics/travel-authorization/pull/57 for examples of previous efforts.

Table name should be pluralized.
User#roles should be transformed from a string to an array of strings.
Roles should be converted to snake case.
Statuses should be converted to snake case.
create_date should be renamed to created_at
add updated_at column

# Implementation Choices

Drop all UserService code, replace it with inline Sequelize model queries.
Update the User model to support unit information, and last sync date with employee directory.
Move /unit endpoint lookup code to /me endpoint and only sync with the government directory once a day.
Upgrade authorization middleware so it returns a Sequelize User model instance.

# Testing Instructions

1. Boot the app via `dev up`.
2. Login to the app at http://localhost:8080/
3. Check that you can see profile information at http://localhost:8080/profile.
4. Go to the "My Travel Requests" page via the top nav.
5. Create a new travel authorization.
6. Go back to the "My Travel Requests" page, check that it loads much faster now that it isn't fetching from the external api all the time.
